packages/core/src/lib/implementation/read-rc-file.ts

```diff
@@ -0,0 +1,51 @@
+import { join } from 'path';
+import {
+  CONFIG_FILE_NAME,
+  CoreConfig,
+  coreConfigSchema,
+} from '@code-pushup/models';
+import { fileExists, importEsmModule } from '@code-pushup/utils';
+
+export class ConfigPathError extends Error {
+  constructor(configPath: string) {
+    super(`Provided path '${configPath}' is not valid.`);
```

Comment:
This error is not applicable to all use cases.

If a use does not enter any configuration, the searching function is called. That internally calls the `readRcByPath` function which does not find anything. But it is not because a provided path is not valid but because the expected default configuration file does not exist. So in this case the user should see something like "Could not find any valid code-pushup.config.(ts|mjs|js) file." or similar.

Reply:
added different error